src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/controllers/web/WorkflowRunController.kt

```diff
@@ -1,37 +1,96 @@
 package org.vaccineimpact.orderlyweb.controllers.web
 
+import com.google.gson.annotations.SerializedName
 import org.vaccineimpact.orderlyweb.ActionContext
+import org.vaccineimpact.orderlyweb.OrderlyServer
+import org.vaccineimpact.orderlyweb.OrderlyServerAPI
+import org.vaccineimpact.orderlyweb.Serializer
 import org.vaccineimpact.orderlyweb.controllers.Controller
+import org.vaccineimpact.orderlyweb.db.AppConfig
 import org.vaccineimpact.orderlyweb.db.repositories.OrderlyWebWorkflowRunRepository
 import org.vaccineimpact.orderlyweb.db.repositories.WorkflowRunRepository
+import org.vaccineimpact.orderlyweb.models.WorkflowRun
+import org.vaccineimpact.orderlyweb.models.WorkflowRunRequest
 import org.vaccineimpact.orderlyweb.models.WorkflowRunSummary
 import org.vaccineimpact.orderlyweb.viewmodels.WorkflowRunViewModel
-import org.vaccineimpact.orderlyweb.models.WorkflowRun
+import java.time.Instant
 
 class WorkflowRunController(
     context: ActionContext,
-    private val workflowRunRepository: WorkflowRunRepository
+    private val workflowRunRepository: WorkflowRunRepository,
+    private val orderlyServerAPI: OrderlyServerAPI
 ) : Controller(context)
 {
-    constructor(context: ActionContext) : this(context, OrderlyWebWorkflowRunRepository())
-
-    fun getWorkflowRunDetails(): WorkflowRun
-    {
-        val key = context.params(":key")
-        return workflowRunRepository.getWorkflowRunDetails(key)
-    }
+    constructor(context: ActionContext) : this(
+        context,
+        OrderlyWebWorkflowRunRepository(),
+        OrderlyServer(AppConfig()).throwOnError()
+    )
 
     @Template("run-workflow-page.ftl")
     fun getRunWorkflow(): WorkflowRunViewModel
     {
         return WorkflowRunViewModel(context)
     }
 
+    fun getWorkflowRunDetails(): WorkflowRun
+    {
+        val key = context.params(":key")
+        return workflowRunRepository.getWorkflowRunDetails(key)
+    }
+
     fun getWorkflowRunSummaries(): List<WorkflowRunSummary>
     {
         return workflowRunRepository.getWorkflowRunSummaries(
             context.queryParams("email"),
             context.queryParams("namePrefix")
         )
     }
+
+    internal data class WorkflowRunResponse(
+        @SerializedName(value = "workflow_key")
+        val key: String,
+        val reports: List<String>
+    )
+
+    fun createWorkflowRun(): String
+    {
+        val workflowRunRequest =
+            Serializer.instance.gson.fromJson(context.getRequestBody(), WorkflowRunRequest::class.java)
+
+        val body = Serializer.instance.gson.toJson(
```

Comment:
So from this I'm guessing that orderly server doesn't like null `changelog` and `ref` values, and we also just need to map 'instances' to 'instance' - is this a genuine plural vs singular issue or just a name mismatch? It would be ideal if these formats were identical but I guess we're not there now - are we moving towards syncing these or is this going to have to be permanent?

Reply:
In the meantime orderly.server has been changed: it now expects `instance` to be a string - which means that the name mismatch is a bit more justifiable. For future-proofing I think it's still worth OW accepting an object i.e. `instances`, which is also consistent with the report-running API.